src/client/tcp.rs

```diff
@@ -26,6 +26,20 @@ pub async fn connect_slave(socket_addr: SocketAddr, slave: Slave) -> Result<Cont
     Ok(context)
 }
 
+/// Connect to a physical, broadcast, or custom Modbus device,
+/// and can enable recover strategy when modbus header mismatch,
+/// when one response frame isn't arrived in specified time due
+/// to tcp retransmission, in the next reading or writing previous
+/// reponse frame will arrived, with this strategy the mismatched
+/// response frame can be discarded. This strategy is inspired by
+/// libmodbus's error recovery: 
+/// https://libmodbus.org/reference/modbus_set_error_recovery/
+pub async fn connect_slave_recover(socket_addr: SocketAddr, slave: Slave, max_recover_retries: usize) -> Result<Context, Error> {
```

Comment:
The number of factory functions for creating a client context gets confusing with all the possible combinations. I guess switching to a builder pattern would be more appropriate, although this is bigger tasks apart from this PR.

@flosse Your thoughts?